### PR TITLE
Bluetooth: Audio - internal changes to media proxy instance structure

### LINKED ISSUE #41201: LE Audio: Improved media_proxy internal data structure
We want an improved internal data structure for the media proxy

the mcc callbacks should be moved to the mprx struct
- 2022-03-02: DONE

the remote player to be guarded bt CONFIG_BT_MCC
- 2022-03-02: DONE - guarded by CONFIG_MCTL_REMOTE_PLAYER_CONTROL_BT

the player pointers to be replaced/changed so that they can be used for player existence checks, and the "registered" variables can be removed.
- 2022-03-03: WON'T DO - will not gain anything in the current implementation.  May become useful for multiple player instances, if the player supplies the player struct.

### PR DESCRIPTION
**Pull request by asbjornsabo**
Updates an include guard, and moves a callback struct inside the media proxy instance struct.

closes https://github.com/zephyrproject-rtos/zephyr/issues/41201

### CODE REVIEW

File: subsys/bluetooth/audio/media_proxy.c
Comment (Thalley) [inline subsys/bluetooth/audio/media_proxy.c:60]: This change in this commit does no longer seem to match the commit description.

File: subsys/bluetooth/audio/media_proxy.c
Comment (Thalley) [inline subsys/bluetooth/audio/media_proxy.c:64]: These callbacks will still rely on `CONFIG_BT_MCC`, so the guard should still be there.

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/media_proxy.c
Comment (asbjornsabo) [reply subsys/bluetooth/audio/media_proxy.c:60]: Commit removed (integrated into next).

File: subsys/bluetooth/audio/media_proxy.c
Comment (asbjornsabo) [reply subsys/bluetooth/audio/media_proxy.c:64]: It is - MCTL_REMOTE_PLAYER_CONTROL depends upon BT_MCC.  
(This is how we have done it elsewhere - only use the "strictest"/"innermost" guard.)

And when MCC is rewritten to register as a client, the callbacks will anyway come with the registration.

**Review by Thalley — COMMENTED** [review]
Seems OK, but won't approve before the depending PR is merged

**Comment by asbjornsabo** [comment]
Rebased on main with no changes.

**Review by Thalley — APPROVED** [review]

**Review by MariuszSkamra — APPROVED** [review]

**Pull request merged**
